### PR TITLE
INTF, change ENUMs to CONSTANTS

### LINKED ISSUE #92: Using ENUMS
Hi,
I have not used enums very much, so I looked into how it works, 

https://blogs.sap.com/2016/10/10/abap-news-release-7.51-enumerations/

notably "Normally, you are not interested in the contents of an enumerated object at all."

in abap-file-formats we are interested in the values, eg setting a json value,

```abap
TYPES:
  basetype TYPE c LENGTH 1,
  BEGIN OF ENUM ver BASE TYPE basetype,
    standard          VALUE IS INITIAL,
    cloud_development VALUE '5',
  END OF ENUM ver.

DATA: BEGIN OF ls_json,
        ver TYPE basetype,
      END OF ls_json.
ls_json-ver = cloud_development.
WRITE ls_json-ver.
```

Which outputs "C" when run

As we are interested in values, probably using structured constants is the better approach?

### PR DESCRIPTION
**Pull request by larshp**
changing ENUMS to CONSTANTS, closes #92

### CODE REVIEW

**Comment by larshp** [comment]
👍

**Comment by schneidermic0** [comment]
The syntax for referencing the constant to the values for a type might not be final, but it gives at least the idea how we would like to solve it.

**Comment by larshp** [comment]
cool, suggest merging

**Comment by schneidermic0** [comment]
> cool, suggest merging

I am just waiting for another approval by @huber-nicolas and/or @wurzka

**Review by schneidermic0 — APPROVED** [review]

File: file-formats/intf/zif_aff_intf_v1.intf.abap
Comment (huber-nicolas) [inline file-formats/intf/zif_aff_intf_v1.intf.abap:3]: Should "Interface category" be the description in the json schema for the type ty_category?
Another option would be to use annotation "$description Interface category"

To give a title to the type you have to write:
\<p class="shorttext synchronized" lang="en"\>This is the title\</p>
or
\<p class="shorttext"\>This is the title\</p>
or
$title This is the title

To make a field required in the schema you have to write:
$required

**Review by huber-nicolas — APPROVED** [review]
I approve, I only made a comment

File: file-formats/intf/zif_aff_intf_v1.intf.abap
Comment (schneidermic0) [reply file-formats/intf/zif_aff_intf_v1.intf.abap:3]: Thanks, Nicolas. Let us add these additions as soon as they are clear from your end in an additional change request so that we can merge this one.

**Pull request merged**
